### PR TITLE
Add end-game VP graph.

### LINKED ISSUE #4275: [Enhancement] Add VP Progression chart (graph) as part of end game results
Display a progression chart of attained VP (total) by player by the end of each generation. 

1. The start of the game, "Generation 0", store starting VP (20)
> Given that the game is at Generation 0" (start of game) 
 > When all the players have made their initial card selection
> Then each played/selected corporation (or player's name) is stored with its initial starting VP 

2. Gained or lost (negative) VP points are added to the previous generation's total for each subsequent generation
> Given I have an initial amount of VP
> When all players have passed the current generation
> Then gained (or lost) VP are reflected in my newly stored VP total for the current generation

3. Chart displayed with result at the end of the game
> Given that terraforming conditions have been met
> And all player's passed I have an initial amount of VP
> When the last players has placed any last potential greenery
> Then the chart with a per generation VP total is displayed with the end of game results.

- See attached image as a sample/idea of add-on

TM_results-VP progression chart

### PR DESCRIPTION
**Pull request by kberg**
image

Fixes #4275

### CODE REVIEW

**Comment by kberg** [comment]
I need to test a couple more things, but this looks decent.

**Comment by kberg** [comment]
(Thank you @nwai90 for the base against which to copy this.)

**Review by nwai90 — APPROVED** [review]

File: src/client/components/gameend/VictoryPointChart.vue [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+        borderColor: ColorStringMap.get(player.color) as string,
+        tension: 0.1,
+        pointRadius: 6,
+      };
+    },
+    getAllPlayerDataSet: function(): Array<ChartDataSet> {
+      return this.players.map((player) => this.getOnePlayerDataSet(player));
+    },
+    renderChart: function(): void {
+      const ctx = document.getElementById('victory-point-chart') as HTMLCanvasElement;
```
Comment (Argeento) [inline src/client/components/gameend/VictoryPointChart.vue:74]:
```suggestion
      const ctx = this.$el.querySelector<HTMLCanvasElement>('#victory-point-chart')! 
```

**Review by Argeento — APPROVED** [review]

**Comment by kberg** [comment]
I'm glad I tested what would happen with legacy games. :boom: Anyway, fixed that.

**Comment by bafolts** [comment]
This is amazing.

**Pull request merged**